### PR TITLE
fix: getHash occasionally panic when rebuild

### PR DESCRIPTION
**Pull request by ahabhgk**
## Test Plan

- add `packages/rspack/tests/watchCases/stats/get-hash-panic`

### CODE REVIEW

File: packages/rspack/tests/watchCases/stats/get-hash-panic/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	plugins: [
+		function (compiler) {
+			compiler.hooks.done.tap("test", stats => {
+				last = stats;
+			});
+		},
+		function (compiler) {
+			compiler.hooks.beforeCompile.tap("test", () => {
+				if (last) {
+					const { hash } = last.toJson({ all: false, hash: true });
```
Comment (9aoy) [inline packages/rspack/tests/watchCases/stats/get-hash-panic/webpack.config.js:15]: fix the panic when rebuild ?

Comment (ahabhgk) [reply packages/rspack/tests/watchCases/stats/get-hash-panic/webpack.config.js:15]: Yes

**Review by 9aoy — APPROVED** [review]

**Comment by h-a-n-a** [comment]
Have you tested the performance regression since you changed lazy to eagerly evaluation?

**Comment by h-a-n-a** [comment]
I thought you don't have to use the compilation from the previous build as it's not necessary....

**Pull request merged**